### PR TITLE
TST: Series repr with pyarrow date32 type

### LINKED ISSUE #34986: BUG: Cannot create third-party ExtensionArrays for datetime types
- [x] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

Code Sample: https://github.com/pandas-dev/pandas/pull/34987

#### Problem description

With https://github.com/pandas-dev/pandas/pull/33400 type checks now fasttrack on the kind of the `dtype`. Thus for every `ExtensionDtype` that is of kind `M`, `is_datetime64tz_dtype` returns `True`.  This is then used in `get_block_type` which leads to the usage of `DatetimeTZBlock` for all of these arrays although this block only supports `DatetimeArray`.

https://github.com/pandas-dev/pandas/blob/4f7d8bba44aa45a81333cdefd7ec627370e7117b/pandas/core/internals/blocks.py#L2685-L2690

### LINKED ISSUE #48238: BUG: series created from pyarrow array with date32 type in 1.5rc0 can be created but not displayed
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd, pyarrow as pa, numpy as np, datetime as dt
arrow_dt = pa.array([dt.date.fromisoformat('2020-01-01')], type=pa.date32())
ser = pd.Series(arrow_dt, dtype=pd.ArrowDtype(arrow_dt.type))
ser

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\core\series.py", line 1594, in __repr__
    return self.to_string(**repr_params)
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\core\series.py", line 1687, in to_string
    result = formatter.to_string()
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 397, in to_string
    fmt_values = self._get_formatted_values()
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 381, in _get_formatted_values
    return format_array(
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 1328, in format_array
    return fmt_obj.get_result()
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 1359, in get_result
    fmt_values = self._format_strings()
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 1833, in _format_strings
    fmt_values = [formatter(x) for x in values]
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 1833, in <listcomp>
    fmt_values = [formatter(x) for x in values]
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 1802, in <lambda>
    return lambda x: _format_datetime64_dateonly(
  File "C:\Users\asafs\Miniconda3\envs\test\lib\site-packages\pandas\io\formats\format.py", line 1792, in _format_datetime64_dateonly
    return x._date_repr
AttributeError: 'datetime.date' object has no attribute '_date_repr'
```

### Issue Description

I saw in the "what's new" for the new release the [functionality](https://pandas.pydata.org/pandas-docs/version/1.5/whatsnew/v1.5.0.html#native-pyarrow-backed-extensionarray) for creating general Arrow-backed objects and wanted to try it out. So I created a simple test Arrow `Array` with `date32` type and tried to make a series from it as described, but the series cannot be displayed without erroring. The `ser` value itself can be created and its array values accessed still.

This issue didn't occur with eg `pa.int32` but I did not comprehensively test across Arrow data types.

And while I'm here, thank you to the team for all your work on the awesome Arrow integration/extension features!

### Expected Behavior

`ser` can be displayed in an interactive REPL as usual and `repr(ser)` does not error.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 224458ee25d92ccdf289d1ae2741d178df4f323e
python           : 3.10.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252

pandas           : 1.5.0rc0
numpy            : 1.23.2
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 60.2.0
pip              : 21.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.31.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           :
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 8.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #48238
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.5.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -2063,3 +1987,11 @@ def test_mode(data_for_grouping, dropna, take_idx, exp_idx, request):
     result = ser.mode(dropna=dropna)
     expected = pd.Series(data_for_grouping.take(exp_idx))
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("box", ["Series", "DataFrame"])
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_arrow.py:1992]: frame_or_series?

Comment (mroeschke) [reply pandas/tests/extension/test_arrow.py:1992]: Thanks, was looking for this fixture

File: pandas/tests/dtypes/test_common.py [hunk 2]
```diff
@@ -239,6 +240,16 @@ def test_is_datetime64tz_dtype():
...
 
 
+def test_custom_ea_kind_M_not_datetime64tz():
+    # GH 34986
+    class NotTZDtype(ExtensionDtype):
+        @property
+        def kind(self) -> str:
+            return "M"
+
+    assert not com.is_datetime64tz_dtype(NotTZDtype())
```
Comment (jbrockmendel) [inline pandas/tests/dtypes/test_common.py:250]: could also check needs_i8_conversion here

**Comment by mroeschke** [comment]
@jbrockmendel don't know if you've ever hit this while working on `EA.argmin/max/sort`. For some reason (seemly) unrelated type's, in this case timedelta, argmin/max/sort fail. I got a similar error when trying to define `ArrowExtensionArray._values_for_argsort`, but ended up removing it since I just could define `ArrowExtensionArray.argmin/max/sort` instead.



https://github.com/pandas-dev/pandas/runs/8027023184?check_suite_focus=true
```
2022-08-26T00:21:38.2857760Z =================================== FAILURES ===================================
2022-08-26T00:21:38.2858430Z _________________ TestIndexReductions.test_minmax_timedelta64 __________________
2022-08-26T00:21:38.2860770Z [gw0] darwin -- Python 3.9.13 /Users/runner/micromamba/envs/test/bin/python
2022-08-26T00:21:38.2861240Z 
2022-08-26T00:21:38.2861500Z obj = array([ 86400000000000, 172800000000000, 259200000000000],
2022-08-26T00:21:38.2862360Z       dtype='timedelta64[ns]')
2022-08-26T00:21:38.2871110Z method = 'argmin', args = (), kwds = {'axis': None, 'out': None}
2022-08-26T00:21:38.2878500Z bound = <built-in method argmin of numpy.ndarray object at 0x178dea3f0>
2022-08-26T00:21:38.2879060Z 
2022-08-26T00:21:38.2879260Z     def _wrapfunc(obj, method, *args, **kwds):
2022-08-26T00:21:38.2892110Z         bound = getattr(obj, method, None)
2022-08-26T00:21:38.2892850Z         if bound is None:
2022-08-26T00:21:38.2893420Z             return _wrapit(obj, method, *args, **kwds)
2022-08-26T00:21:38.2894830Z     
2022-08-26T00:21:38.2896420Z         try:
2022-08-26T00:21:38.2896970Z >           return bound(*args, **kwds)
2022-08-26T00:21:38.2898150Z E           TypeError: Cannot cast array data from dtype('<m8[ns]') to dtype('<m8[us]') according to the rule 'safe'
2022-08-26T00:21:38.2898680Z 
2022-08-26T00:21:38.2899440Z ../../../micromamba/envs/test/lib/python3.9/site-packages/numpy/core/fromnumeric.py:57: TypeError
2022-08-26T00:21:38.2899890Z 
2022-08-26T00:21:38.2900210Z During handling of the above exception, another exception occurred:
2022-08-26T00:21:38.2900610Z 
2022-08-26T00:21:38.2901140Z self = <pandas.tests.reductions.test_reductions.TestIndexReductions object at 0x15efe5d00>
2022-08-26T00:21:38.2901860Z 
2022-08-26T00:21:38.2902170Z     def test_minmax_timedelta64(self):
2022-08-26T00:21:38.2902670Z     
2022-08-26T00:21:38.2903110Z         # monotonic
2022-08-26T00:21:38.2903720Z         idx1 = TimedeltaIndex(["1 days", "2 days", "3 days"])
2022-08-26T00:21:38.2905060Z         assert idx1.is_monotonic_increasing
2022-08-26T00:21:38.2905630Z     
2022-08-26T00:21:38.2906250Z         # non-monotonic
2022-08-26T00:21:38.2906710Z         idx2 = TimedeltaIndex(["1 days", np.nan, "3 days", "NaT"])
2022-08-26T00:21:38.2907330Z         assert not idx2.is_monotonic_increasing
2022-08-26T00:21:38.2907860Z     
2022-08-26T00:21:38.2909520Z         for idx in [idx1, idx2]:
2022-08-26T00:21:38.2911780Z             assert idx.min() == Timedelta("1 days")
2022-08-26T00:21:38.2912570Z             assert idx.max() == Timedelta("3 days")
2022-08-26T00:21:38.2913080Z >           assert idx.argmin() == 0
2022-08-26T00:21:38.2913940Z 
2022-08-26T00:21:38.2914320Z pandas/tests/reductions/test_reductions.py:266: 
2022-08-26T00:21:38.2915080Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-08-26T00:21:38.2915810Z pandas/core/indexes/base.py:7175: in argmin
2022-08-26T00:21:38.2916930Z     return super().argmin(skipna=skipna)
2022-08-26T00:21:38.2917250Z pandas/core/base.py:713: in argmin
2022-08-26T00:21:38.2917510Z     return delegate.argmin()
2022-08-26T00:21:38.2917860Z pandas/core/arrays/_mixins.py:200: in argmin
2022-08-26T00:21:38.2918570Z     return nargminmax(self, "argmin", axis=axis)
2022-08-26T00:21:38.2919230Z pandas/core/sorting.py:480: in nargminmax
2022-08-26T00:21:38.2920470Z     return _nanargminmax(arr_values, mask, func)
2022-08-26T00:21:38.2920970Z pandas/core/sorting.py:491: in _nanargminmax
2022-08-26T00:21:38.2921390Z     return non_nan_idx[func(non_nans)]
2022-08-26T00:21:38.2921820Z <__array_function__ internals>:180: in argmin
2022-08-26T00:21:38.2922200Z     ???
2022-08-26T00:21:38.2923120Z ../../../micromamba/envs/test/lib/python3.9/site-packages/numpy/core/fromnumeric.py:1312: in argmin
2022-08-26T00:21:38.2923920Z     return _wrapfunc(a, 'argmin', axis=axis, out=out, **kwds)
2022-08-26T00:21:38.2924780Z ../../../micromamba/envs/test/lib/python3.9/site-packages/numpy/core/fromnumeric.py:66: in _wrapfunc
2022-08-26T00:21:38.2926060Z     return _wrapit(obj, method, *args, **kwds)
2022-08-26T00:21:38.2927600Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-08-26T00:21:38.2927880Z 
2022-08-26T00:21:38.2928170Z obj = array([ 86400000000000, 172800000000000, 259200000000000],
2022-08-26T00:21:38.2928850Z       dtype='timedelta64[ns]')
2022-08-26T00:21:38.2929530Z method = 'argmin', args = (), kwds = {'axis': None, 'out': None}
2022-08-26T00:21:38.2931450Z wrap = <built-in method __array_wrap__ of numpy.ndarray object at 0x178dea3f0>
2022-08-26T00:21:38.2935490Z 
2022-08-26T00:21:38.2935730Z     def _wrapit(obj, method, *args, **kwds):
2022-08-26T00:21:38.2936160Z         try:
2022-08-26T00:21:38.2936550Z             wrap = obj.__array_wrap__
2022-08-26T00:21:38.2941120Z         except AttributeError:
2022-08-26T00:21:38.2957840Z             wrap = None
2022-08-26T00:21:38.2958820Z >       result = getattr(asarray(obj), method)(*args, **kwds)
2022-08-26T00:21:38.2962350Z E       TypeError: Cannot cast array data from dtype('<m8[ns]') to dtype('<m8[us]') according to the rule 'safe'
2022-08-26T00:21:38.2962810Z 
2022-08-26T00:21:38.2963520Z ../../../micromamba/envs/test/lib/python3.9/site-packages/numpy/core/fromnumeric.py:43: TypeError
```

**Comment by jbrockmendel** [comment]
i dont think ive seen that, no.  is it flaky?

**Comment by jbrockmendel** [comment]
[finger slipped on the wrong button]

**Comment by mroeschke** [comment]
> i dont think ive seen that, no. is it flaky?

I don't think so since it appears on some specific builds. I've tried installing those environments on my laptop but can replicating them.

I'll keep poking around. I suspect it has to do with some statefulness.

**Comment by mroeschke** [comment]
Noting if worked on the future: Disabling the pyarrow timedelta types in the tests fixed the seemingly unrelated timedelta errors in https://github.com/pandas-dev/pandas/pull/48258#issuecomment-1239860302, so there may be some odd interaction between then for pyarrow version 5, 6, 7

**Comment by mroeschke** [comment]
May revisit this later, but since the original issue has been address by another PR, closing

**Pull request closed without merging**